src/mantine-core/src/components/Stepper/Step/Step.tsx

```diff
@@ -95,6 +99,12 @@ export const Step = forwardRef<HTMLButtonElement, StepProps>(
     const _iconSize = theme.fn.size({ size, sizes: defaultIconSizes });
     const _icon = state === 'stepCompleted' ? null : state === 'stepProgress' ? progressIcon : icon;
 
+    useEffect(() => {
+      if (['stepProgress', 'stepCompleted'].includes(state)) {
+        onStepReached && onStepReached();
```

Comment:
I think there is no need for this prop as Stepper is always controlled

Reply:
Good point, fixed.